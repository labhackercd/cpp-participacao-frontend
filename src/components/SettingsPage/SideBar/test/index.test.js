import React from 'react';
import ReactDOM from 'react-dom';
import MockTheme from '../../../mockTheme'

import {shallow, render, mount} from "enzyme";
import SettingsSideBar from './../index';

it("snapshot should not have changes", () => {
    const component = shallow(<SettingsSideBar/>);
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});

test('Test if NavBar renders without crash', () => {
    const div = document.createElement("div")
    ReactDOM.render(<SettingsSideBar/>, div)
    ReactDOM.unmountComponentAtNode(div)
});

test('Test if clicking on item of sidebar ', () => {
    const wrapper = mount(<SettingsSideBar />);
    const itemButton = wrapper.find("#tools").last();
    itemButton.simulate('click');
});

