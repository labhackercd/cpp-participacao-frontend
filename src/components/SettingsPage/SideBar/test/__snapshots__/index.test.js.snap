// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should not have changes 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-homeMenuContainer-2"
  container={true}
>
  <WithStyles(ForwardRef(Drawer))
    PaperProps={
      Object {
        "style": Object {
          "backgroundColor": "transparent",
          "boxShadow": "none",
        },
      }
    }
    classes={
      Object {
        "paper": "makeStyles-drawerPaper-1",
      }
    }
    variant="permanent"
  >
    <WithStyles(ForwardRef(List))>
      <ListItemSideBar
        classes={
          Object {
            "drawerPaper": "makeStyles-drawerPaper-1",
            "homeMenuContainer": "makeStyles-homeMenuContainer-2",
          }
        }
        icon={
          Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "displayName": "SettingsIcon",
            "muiName": "SvgIcon",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          }
        }
        id="generalSettings"
        isActive={true}
        itemId={0}
        title="Configurações Gerais"
      />
      <ListItemSideBar
        classes={
          Object {
            "drawerPaper": "makeStyles-drawerPaper-1",
            "homeMenuContainer": "makeStyles-homeMenuContainer-2",
          }
        }
        icon={
          Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "displayName": "SettingsIcon",
            "muiName": "SvgIcon",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          }
        }
        id="tools"
        isActive={false}
        itemId={1}
        title="Ferramentas"
      />
      <ListItemSideBar
        classes={
          Object {
            "drawerPaper": "makeStyles-drawerPaper-1",
            "homeMenuContainer": "makeStyles-homeMenuContainer-2",
          }
        }
        icon={
          Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "displayName": "InfoIcon",
            "muiName": "SvgIcon",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          }
        }
        id="about"
        isActive={false}
        itemId={2}
        title="Sobre o Dashboard"
      />
    </WithStyles(ForwardRef(List))>
  </WithStyles(ForwardRef(Drawer))>
</WithStyles(ForwardRef(Grid))>
`;
